src/main/java/com/meilisearch/sdk/json/GsonKeyTypeAdapter.java

```diff
@@ -0,0 +1,191 @@
+package com.meilisearch.sdk.json;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import com.meilisearch.sdk.model.Key;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class GsonKeyTypeAdapter extends TypeAdapter<Key> {
+
+    private static final String KEY_NAME = "name";
+    private static final String KEY_DESCRIPTION = "description";
+    private static final String KEY_UID = "uid";
+    private static final String KEY_KEY = "key";
+    private static final String KEY_ACTIONS = "actions";
+    private static final String KEY_INDEXES = "indexes";
+    private static final String KEY_EXPIRES_AT = "expiresAt";
+    private static final String KEY_CREATED_AT = "createdAt";
+    private static final String KEY_UPDATED_AT = "updatedAt";
+    private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+
+    @Override
+    public void write(JsonWriter writer, Key key) throws IOException {
+        boolean serializeNulls = writer.getSerializeNulls();
+        writer.setSerializeNulls(true);
+
+        try {
+            writeStartObject(writer);
+            writeString(writer, KEY_NAME, key.getName());
+            writeString(writer, KEY_DESCRIPTION, key.getDescription());
+            writeString(writer, KEY_UID, key.getUid());
+            writeString(writer, KEY_KEY, key.getKey());
+            writeStringArray(writer, KEY_ACTIONS, key.getActions());
+            writeStringArray(writer, KEY_INDEXES, key.getIndexes());
+            if (key.getExpiresAt() != null) {
+                writeDate(writer, KEY_EXPIRES_AT, key.getExpiresAt());
+            } else {
+                writeNull(writer, KEY_EXPIRES_AT);
+            }
+            writeDate(writer, KEY_CREATED_AT, key.getCreatedAt());
+            writeDate(writer, KEY_UPDATED_AT, key.getUpdatedAt());
+            writeEndObject(writer);
+        } finally {
+            writer.setSerializeNulls(serializeNulls);
+        }
+    }
+
+    @Override
+    public Key read(JsonReader reader) throws IOException {
+        return nullOrElse(
+                reader,
+                r -> {
+                    Key key = new Key();
+                    readStartObject(r);
+                    while (r.peek() != JsonToken.END_OBJECT) {
```

Comment:
Are you really obligated to define all the keys of the object? Can't you just change the ones you don't like the default shape?

Reply:
Can we have something more dynamic, because in the future releases that we change/include new items into the keys object we will have to remember to update this file as well :)

Reply:
> Are you really obligated to define all the keys of the object? Can't you just change the ones you don't like the default shape?

I couldn't find a way to do that, unfortunately. That is what I wanted to do at first. But it seems that when you define a type adapter like this one, Gson will just call your code and I couldn't find a way to fall back to some kind of default implementation for some parts.

I also tried just writing a serializer but I couldn't get it working (I can't remember why now as it's been a little while since I worked on this)

> Can we have something more dynamic, because in the future releases that we change/include new items into the keys object we will have to remember to update this file as well :)

I totally understand where you're coming from. I don't think this approach is ideal. I actually prefer the approach that you already have in the project (though there's a few small things I'd change) because I think it allows for a lighter touch in maintainence. However, it seems that the purpose behind this ticket is to remove that Key specific logic from the encode method in the `GsonJsonHandler` and this felt like the most logical way to do that.

I can always spend some more time trying out some different approaches, if you'd prefer a different method that is more maintanable?

Reply:
No no, I think it is ok the way it is, I just asked to ensure I extract every bit of information ;)